command_line/powder_calibrate.py

```diff
@@ -723,15 +697,52 @@ def calibrate_with_calibrant(
         # update geometry and save to .expt
         ai = gonio_geom.get_ai()
         self.geometry.update_from_ai(ai)
-        self.geometry.save_to_expt(output=self.user_args.calibrated_geom)
+        self.geometry.save_to_expt(output=self.output.calibrated_geom)
 
         logger.info(f"Geometry fitted by pyFAI:\n----- \n {self.geometry} \n")
         self.show_straight_lines(ai, show=plots)
 
 
 @dials.util.show_mail_handle_errors()
-def run(args=None):
-    calibrator = PowderCalibrator()
+def run(args: List[str] = None, phil: scope = phil_scope) -> None:
+    """
+    Check command-line input and do the sequence of
+    function calls.
+
+    :param args: The arguments supplied by the user (default: sys.argv[1:])
+    :param phil: The PHIL scope definition (default: phil_scope, the master PHIL scope
+    for this program).
+    """
+    usage = "$ dials.powder_calibrate imported.expt standard=standard_name [options]"
+
+    parser = ArgumentParser(
+        usage=usage, phil=phil, read_experiments=True, check_format=True, epilog=__doc__
+    )
+
+    parameters, options = parser.parse_args(args=args, show_diff_phil=False)
+
+    # Configure the logging.
+    dials.util.log.config(options.verbose, logfile=parameters.output.log)
+
+    # Log the dials version
+    logger.info(dials_version())
+
+    # Log the difference between the PHIL scope definition and the active PHIL scope,
+    # which will include the parsed user inputs.
+    diff_phil = parser.diff_phil.as_str()
+    if diff_phil:
+        logger.info("The following parameters have been modified:\n%s", diff_phil)
+
```

Comment:
If run without command-line arguments it's nice for the program to provide basic help:
```suggestion
    if len(experiments) == 0:
        parser.print_help()
        return
```

Reply:
Similarly, `standard="something"` is a mandatory argument. With the default `standard=None` the program exits with a stacktrace. Could this be a nice apologetic message instead?

Reply:
I had a dream that I did cover those bases. I think I have officially looked too long at this code.